File: 7-kyu/complete-the-pattern-#7---cyclical-permutation/python/solution.py
def pattern(n):
    ret = []
    for i in range(1, n + 1):
        to_append = ''
        to_append += ''.join(str(j) for j in range(i, n + 1))
        to_append += ''.join(str(j) for j in range(1, i))
        ret.append(to_append)
    return '\n'.join(ret)

File: 7-kyu/complete-the-pattern-#7---cyclical-permutation/c++/solution.cpp
#include <string>

std::string pattern(int n)
{
    std::string ret = "";
    if (n < 1) return ret;
    if (n == 1) return "1";
    for (unsigned i = 1; i <= n; i++) {
        std::string to_append = "";
        for (unsigned j = i; j <= n; j++) to_append += std::to_string(j);
        for (unsigned j = 1; j < i; j++) to_append += std::to_string(j);
        ret += to_append + "\n";
    }
    ret.pop_back();
    return ret;
}